storage/innobase/handler/ha_innodb.cc

```diff
@@ -1116,6 +1116,42 @@ static SHOW_VAR innodb_status_variables[]= {
   {NullS, NullS, SHOW_LONG}
 };
 
+/** Deinitialize InnoDB statistics when closing the last table handle. */
+class InnoDB_share final : public Handler_share
+{
+  /** InnoDB table that contains persistent statistics */
+  dict_table_t &table;
```

Comment:
The data member needs to be a pointer and not a reference. On table-rebuilding operations or when a `DROP TABLE` operation is committed, we must update this to point to the new table or invalidate this to `nullptr`. Currently, there are plenty of `heap-use-after-free` errors in table-rebuilding tests.